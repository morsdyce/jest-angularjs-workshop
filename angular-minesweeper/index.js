import 'common/assets/stylesheets/minefield.css';

import App from 'angular-minesweeper/components/app/app.module';
import Common from 'angular-minesweeper/common/common.module';
import Board from 'angular-minesweeper/components/board/board.module';

export default angular.module('ngReact', [
  Common.name,
  Board.name,
  App.name
]);