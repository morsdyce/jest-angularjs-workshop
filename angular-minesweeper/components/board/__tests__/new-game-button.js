import 'angular-minesweeper';
import { noop } from 'lodash';

describe('<new-game-button>', () => {
  let createComponent, $scope, element;

  beforeEach(angular.mock.module('ngReact.board'));

  beforeEach(inject(($rootScope, $compile) => {
    createComponent = (html, scopeAttrs) => {
      $scope = angular.extend($rootScope.$new(), scopeAttrs);

      element = angular.element(html);
      element = $compile(element)($scope);
      $scope.$digest();
    };
  }));

  test('should render a new game button', () => {
    const html = `<new-game-button new-game="newGame"></new-game-button>`;
    createComponent(html, { newGame: noop });

    expect(element.html()).toMatchSnapshot();
  });

  test('should call new game function when clicking on the button', () => {
    const html = `<new-game-button new-game="newGame"></new-game-button>`;
    const newGameSpy = jest.fn();
    createComponent(html, { newGame: newGameSpy });

    expect(newGameSpy).not.toHaveBeenCalled()

    element.find('img').triggerHandler('click');

    expect(newGameSpy).toHaveBeenCalled()
  });

});

