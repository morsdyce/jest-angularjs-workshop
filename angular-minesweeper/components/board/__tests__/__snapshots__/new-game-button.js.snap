// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<new-game-button> should render a new game button 1`] = `
"<div class=\\"new-game-button\\">
  <img ng-src=\\"newgame.jpg\\" ng-click=\\"NewGameButton.startNewGame()\\" src=\\"unsafe:newgame.jpg\\">
</div>
"
`;
